background color by data in table widget

broncha opened the pull request:
@grassick I implemented this for each column, adding a separate `backgroundColorAxis` which includes the expression from textAxis and only xform and colormap from the `backgroundColorAxis` is used.
Added `showExpr` to `AxisComponent` as expression would be duplicated for the column!

@Autio here is how the implementation looks. 
<img width="676" alt="Screen Shot 2022-10-28 at 15 51 55" src="https://user-images.githubusercontent.com/46905/198562885-9812ead9-b82d-4576-91de-eb3205170937.png">
<img width="512" alt="Screen Shot 2022-10-28 at 15 51 59" src="https://user-images.githubusercontent.com/46905/198562910-83cf5d63-e064-48f5-92df-3d9d1644aaf6.png">
<img width="610" alt="Screen Shot 2022-10-28 at 15 52 03" src="https://user-images.githubusercontent.com/46905/198562928-cfa41f05-e544-40a3-9d6b-11a984a41190.png">

grassick commented:
There is some complexity around aggregate columns that isn't being taken care of. If I:
- make a table of water points
- set first column to name
- set second to location accuracy
- add background color to second axis by range 


I get: 
![image](https://user-images.githubusercontent.com/1238335/198582658-05216b0a-1af8-4e07-b337-6074f7bdf2fa.png)

Also, it seems to default the backgroundColorAxis to the textAxis, but then doesn't keep them in sync thereafter. But there's no way to edit the expression of the backgroundColorAxis!

Maybe the two axes should be independent? @Autio ? That is, you can color by something that isn't the same expression as the textAxis. This would get rid of the need for `showExpr`

Also, note that the textAxis is really just using the expression inside the axis, not the entire axis, for weird historical reasons.

The `aggrNeed: "required"` in backgroundColorAxis is going to cause you trouble, as it doesn't always need to be aggregate. But if it is, the table compiler should take it into account and add groupings to all other fields.

broncha commented:
@grassick I removed showExpr in AxisComponent, fixed the grouping in background color axis query. What do you suggest for aggrNeed? I have removed that also for now.
And color axis is no longer dependant on the textAxis

grassick commented:
This is great @broncha ! I added aggrNeed: "optional" and it works perfectly. Merging it in!
